### PR TITLE
Problem: sdk 0.50 is not used

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by mmsqe — COMMENTED** [review]
do we need regen protos with cosmossdk.io for customtype?

**Comment by yihuang** [comment]
> do we need regen protos with cosmossdk.io for customtype?

yes, still downloading docker images.

File: go.mod [hunk 1]
```diff
@@ -1,223 +1,246 @@
...
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	nhooyr.io/websocket v1.8.6 // indirect
-	pgregory.net/rapid v0.5.5 // indirect
+	gotest.tools/v3 v3.5.1 // indirect
+	nhooyr.io/websocket v1.8.10 // indirect
+	pgregory.net/rapid v1.1.0 // indirect
 )
 
 replace (
+	cosmossdk.io/collections => github.com/mmsqe/cosmos-sdk/collections v0.0.0-20240307052158-b7a6ec132cd3
```
Comment (mmsqe) [inline go.mod]: can wait [this](https://github.com/cosmos/cosmos-sdk/pull/19675) fix a bit

File: go.mod [hunk 2]
```diff
@@ -1,223 +1,250 @@
...
 )
 
 replace (
 	// use cosmos keyring
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.2.0
-	github.com/cometbft/cometbft => github.com/cometbft/cometbft v0.37.2-0.20230905115601-790d57e1748f
-	github.com/cometbft/cometbft-db => github.com/crypto-org-chain/cometbft-db v0.0.0-20231011055109-57922ac52a63
-	github.com/cosmos/cosmos-sdk => github.com/crypto-org-chain/cosmos-sdk v0.46.0-beta2.0.20231013072415-eec017435467
+	github.com/cockroachdb/pebble => github.com/cockroachdb/pebble v0.0.0-20230209160836-829675f94811
+	github.com/cosmos/cosmos-sdk => github.com/mmsqe/cosmos-sdk v0.46.0-beta2.0.20240315010123-8531387f70f0
```
Comment (mmsqe) [inline go.mod]: test eth priority pass, can wait the [fix](https://github.com/cosmos/cosmos-sdk/pull/19759) a bit

**Review by mmsqe — APPROVED** [review]

File: x/evm/keeper/keeper.go [hunk 3]
```diff
@@ -18,13 +18,14 @@ package keeper
 import (
 	"math/big"
 
+	corestoretypes "cosmossdk.io/core/store"
 	errorsmod "cosmossdk.io/errors"
-	"github.com/cometbft/cometbft/libs/log"
+	"cosmossdk.io/log"
+	sdkmath "cosmossdk.io/math"
```
Comment (yihuang) [inline x/evm/keeper/keeper.go]:
```suggestion
```

**Pull request merged**